### PR TITLE
Runtime: Using lamba version of AddService when adding MemoryStorage.

### LINKED ISSUE #5325: Runtime: In a Azure Functions project, the wrong constructor is selected via DI
MemoryStorage in Runtime was being added via:

```
services.AddSingleton<IStorage, MemoryStorage>();
```

When running in a Azure Functions project, the incorrect constructor for MemoryStorage was being selected:

```
public MemoryStorage(JsonSerializer jsonSerializer, Dictionary<string, JObject> dictionary = null)
```
Except with a null 'jsonSerializer', causing it to throw.

This does not happen in a WebApp project.

### PR DESCRIPTION
**Pull request by tracyboehrer**
Fixes #5325

### CODE REVIEW

File: libraries/integration/Microsoft.Bot.Builder.Integration.Runtime/Extensions/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -116,7 +116,7 @@ internal static void AddBotRuntimeStorage(this IServiceCollection services, ICon
             else
             {
                 // If no storage is configured, default to memory storage
-                services.AddSingleton<IStorage, MemoryStorage>();
+                services.AddSingleton<IStorage>(sp => new MemoryStorage());
```
Comment (johnataylor) [inline libraries/integration/Microsoft.Bot.Builder.Integration.Runtime/Extensions/ServiceCollectionExtensions.cs:119]: why did you find you had to do this?

Comment (tracyboehrer) [reply libraries/integration/Microsoft.Bot.Builder.Integration.Runtime/Extensions/ServiceCollectionExtensions.cs:123]: Me, Carlos, Gabo, and Peter all tried to trace down why this was happening, and only in an Azure Function project.  We could find no definitive reason why DI was selecting the wrong constructor.

Comment (tracyboehrer) [reply libraries/integration/Microsoft.Bot.Builder.Integration.Runtime/Extensions/ServiceCollectionExtensions.cs:123]: At this point, this is one of two work arounds I can devise.  This, or change MemoryStorage to default the JsonSerializer if a null is provided via DI.  By all appearances, something in Azure Functions is adding the null JsonSerializer.

**Comment by tracyboehrer** [comment]
Closing this PR.  The issue will be resolved when we switch to AdaptiveDialogBot in another PR.

**Pull request closed without merging**